app/components/CoinForm.tsx

```diff
@@ -0,0 +1,279 @@
+    <div className="drop-shadow-xl">
+      <div
+        style={{ width: 700, height: 300 }}
+        className="flex justify-center items-center flex-col bg-accent rounded-lg m-10  p-10"
+      >
+        <div className="flex items-center justify-between w-full">
+          <div className="text-second">Select Coins</div>
+          <div className="w-16 flex justify-between mb-3">
+            <button onClick={resetForm}>
+              <ResetIcon />
+            </button>
+            <button onClick={closeForm}>
+              <CloseIcon />
+            </button>
+          </div>
+        </div>
+        <div className=" w-full flex justify-center items-center ">
+          <div className="h-48 w-1/2 flex items-center justify-center p-10 ">
+            {coin.name && (
+              <div className="relative ">
+                <div class="absolute -inset-5">
+                  <div class="w-full h-full max-w-sm mx-auto lg:mx-0 opacity-30 blur-lg bg-gradient-to-r from-second to-primary"></div>
+                </div>
+                <div className="flex items-center justify-center relative p-8 text-lg font-bold text-second bg-accent font-pj rounded-xl ">
+                  <img src={coin.image} className="h-16" />
+                  {coin.name}
+                </div>
+              </div>
+            )}
+          </div>
+          <div className="flex flex-col justify-start h-48 w-1/2 items-start ">
+            <form>
+              {missingCoin && (
+                <p
+                  role="alert"
+                  className="text-xs"
+                  style={{ color: "rgb(200, 0, 0)" }}
+                >
+                  Please choose a coin.
+                </p>
+              )}
+              <input
+                value={searchValue ? searchValue : coin.name || ""}
+                onChange={handleSearchChange}
+                onKeyDown={handleKeyPress}
+                placeholder={"Start typing to find your coin..."}
+                type="text"
+                className={`mx-3 drop-shadow-md rounded-sm pl-3 w-72 shadow-md ${
+                  missingCoin ? "border-2 mb-2 border-rose-600" : "my-3"
+                }`}
+              />
+              <div className="absolute">
+                {searchValue &&
+                  currentCoins?.map((item) => {
+                    const name = item.name.toLowerCase();
+                    const search = searchValue.toLowerCase();
+                    if (name.startsWith(search))
+                      return (
+                        <div key={item.id} className="border-slate-300">
+                          <DropdownRow
+                            key={item.id}
+                            className="bg-second"
+                            onClick={() => handleSearch(item)}
+                          >
+                            {item.name}
+                          </DropdownRow>
+                        </div>
+                      );
+                  })}
```

Comment:
you need to filter before you map

Reply:
it should be done like this






       currentCoins?
         .filter((item) => {
            const name = item.name.toLowerCase();
            const search = searchValue.toLowerCase();
            return name.startsWith(search)
         })
         .map(item => (
            <div key={item.id} className="border-slate-300">
                <DropdownRow
                  key={item.id}
                  className="bg-second"
                  onClick={() => handleSearch(item)}
                >
                  {item.name}
                </DropdownRow>
            </div>
         ))

Reply:
I'll ask about this one in the call